tests/integration/test_s3.py

```diff
@@ -947,70 +972,45 @@ def test_set_external_hostname(self):
         finally:
             config.HOSTNAME_EXTERNAL = hostname_before
 
-    def test_s3_website_errordocument(self):
-        # check that the error document is returned when configured
-        bucket_name = 'test-bucket-%s' % short_uid()
-        client = self._get_test_client()
+    def test_s3_static_website_hosting(self):
+        bucket_name = 'test-%s' % short_uid()
 
-        client.create_bucket(Bucket=bucket_name)
-        client.put_object(Bucket=bucket_name, Key='error.html', Body='This is the error document')
-        client.put_bucket_website(
+        self.s3_client.create_bucket(Bucket=bucket_name)
+        self.s3_client.put_object(Bucket=bucket_name, Key='test/index.html', Body='index')
+        self.s3_client.put_object(Bucket=bucket_name, Key='test/error.html', Body='error')
+        self.s3_client.put_object(Bucket=bucket_name, Key='actual/key.html', Body='key')
+        self.s3_client.put_bucket_website(
             Bucket=bucket_name,
-            WebsiteConfiguration={'ErrorDocument': {'Key': 'error.html'}}
-        )
-        url = client.generate_presigned_url(
-            'get_object', Params={'Bucket': bucket_name, 'Key': 'nonexistent'}
+            WebsiteConfiguration={'IndexDocument': {'Suffix': 'index.html'},
+                'ErrorDocument': {'Key': 'test/error.html'}}
         )
-        response = requests.get(url, verify=False)
-        self.assertEqual(response.status_code, 200)
-        self.assertEqual(response.text, 'This is the error document')
-        # cleanup
-        client.delete_object(Bucket=bucket_name, Key='error.html')
-        client.delete_bucket(Bucket=bucket_name)
 
-        # check that normal responses are returned for bucket with index configuration, but not error document
-        bucket_name = 'test-bucket-%s' % short_uid()
-        client.create_bucket(Bucket=bucket_name)
-        client.put_bucket_website(
-            Bucket=bucket_name,
-            WebsiteConfiguration={'IndexDocument': {'Suffix': 'index.html'}}
-        )
-        url = client.generate_presigned_url(
-            'get_object', Params={'Bucket': bucket_name, 'Key': 'nonexistent'}
-        )
-        response = requests.get(url, verify=False)
-        self.assertEqual(response.status_code, 404)
-        # cleanup
-        client.delete_bucket(Bucket=bucket_name)
+        headers = aws_stack.mock_aws_request_headers('s3')
+        headers['Host'] = '{}.{}:4566'.format(bucket_name, constants.S3_STATIC_WEBSITE_HOSTNAME)
```

Comment:
nit: better use `config.EDGE_PORT` here instead of hardcoding port `4566`. (Same in lines 992, 1004, 1010 below).